lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -546,6 +552,15 @@ instance Arbitrary ApiFee where
 instance Arbitrary AddressPoolGap where
     arbitrary = arbitraryBoundedEnum
 
+instance Arbitrary (Iso8601Range (name :: Symbol)) where
+    arbitrary = do
+        (mt1, mt2) <- arbitrary
+        let (mv1, mv2) =
+                if all isJust [mt1, mt2]
+                then (min mt1 mt2, max mt1 mt2)
```

Comment:
Not sure to get this one? This seems to be forbidding `mt1` to be greater than `mt2` :thinking: ? (whereas that's a valid case for the range..)

Reply:
> Not sure to get this one? This seems to be forbidding `mt1` to be greater than `mt2` thinking ? (whereas that's a valid case for the range..)

Hmm. I don't see how that's a valid case.

From my understanding of `Iso8601Range v1 v2`:

* `v1` is an optional _earliest time_.
* `v2` is an optional _latest time_.

So this `Arbitrary` instance is designed to only generate the following cases:

* `Iso8601Range Nothing Nothing`
* `Iso8601Range Nothing (Just t)`
* `Iso8601Range (Just t) Nothing`
* `Iso8601Range (Just t1) (Just t2)` where `(t1 <= t2)`.

Perhaps I have misunderstood?

Reply:
> Not sure to get this one? This seems to be forbidding `mt1` to be greater than `mt2` thinking ? (whereas that's a valid case for the range..)

I've just read your later comment. I'll update the code to support both ascending and descending ranges.

Reply:
@KtorZ Now updated to allow ranges in both ascending and descending order.